### DISCUSSION TITLE
Specify VARCHAR Length On @belongsTo/hasOne

### DISCUSSION THREAD

**Discussion by A-Corregidor** [loopbackio/loopback-next#8680] [Q&A]
Hi there!

When you create some model relation field (userId), using these syntax

```
@belongsTo(() => User)
	userId: string;
```

It's created on DB like a VARCHAR(512) (at least, on MySQL).
How could I modify that length? My UUID's real length is 36

Thanks

**Comment by A-Corregidor** [comment]
Possible referenced question: https://github.com/loopbackio/loopback-next/discussions/8388

**Comment by achrinza** [comment]
Hi @A-Corregidor, does https://github.com/loopbackio/loopback-next/discussions/8476#discussioncomment-2543550 help answer your question?

In this case, it should be something like this:

```typescript
@belongsTo(
  () => User,
  undefined,
  {
    mysql: {
      dataType: 'varchar',
      dataLength: 36,
    },
  },
)
userId: number;
```

**Reply by A-Corregidor** [reply]
Hi @achrinza , it doesn't seems to work.
If i set that code and run a (soft) migration, userId keeps being a VARCHAR(512).
It works if I set it like
```
@belongsTo(
  () => User,
  undefined,
  {
    length: 36
  },
)
userId: number;
```

But then, the foreign key fails as id on User's table is VARCHAR(255) (which I would like to change too, but I don't know how) and userId is VARCHAR(36) (see https://github.com/loopbackio/loopback-next/discussions/8388)

**Reply by achrinza** [reply]
Thanks for the additional information. I suspect that this is an issue with the `autoupdate` logic in the MySQL connector; In this case, would you like to open an issue in that GitHub repo?

I'll also put this issue in my backlog, but it'll be some time before I can get to it. Though we're also open to accepting patches from the community as well.
